enhancements/olm-managed-metrics.md

```diff
@@ -0,0 +1,146 @@
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   └── example.rbac.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC files. This enhancement proposes expanding this feature to support `Namespaces`, `ServiceMonitors` and `PrometheusRules` objects:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   ├── example.ns.yaml
+    │   ├── example.prometheusrule.yaml
+    │   ├── example.rbac.yaml
+    │   └── example.servicemonitor.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+With this change, OLM will need to be updated to manage the `Namespace`, `ServiceMonitor` and `PrometheusRule` objects through the lifecycle of an operator.
+
+#### Namespace Support
+
+As OLM gains support for `Namespace` objects, The `ClusterServiceVersion` will be updated to include an optional `NamespaceRequirements` object field. The `NamespaceRequirements` will look like the following:
+
+```golang
+type NamespaceRequirements struct {
+  //+optional
+  // name represents where the operator must be deployed.
+  name string
+
+  //+optional
+  // labels are labels that must be present on the given namespace.
+  labels []string
+}
+```
+
+>Note: If `NamespaceRequirements.namespace` is set, the UI will only show the specified namespace as an install option.
+
+#### ServiceMonitor and PrometheusRule Support
+
+As OLM gains support for these Prometheus objects, there is a chance that installed operators may overload the Prometheus instance. It is important that a Cluster Admin is alerted that an operator will report metrics prior to deploying the operator. As such, we propose adding a `monitoringSupported` boolean field to the `ClusterServiceVersion` object. OLM will not install either of the Prometheus object unless the `monitoringSupported` field is set to true. Additionally, Cluster Admins will need to manually the operator's `InstallPlans` in the following scenarios:
+
+- A fresh install for this operator is being created where this field is set to `true`
+- An existing operator had this field set to `false` and is being upgraded to a new version that sets this field to `true`
+
+> Note: The `InstallPlan` will include a helpful message for Cluster Admins if either of the situations above are encountered.
+
+### OLM Telemeter Support
+
+#### Telemeter Requirements
+
+Once OLM is capable of managing `ServiceMonitor` and `PrometheusRule` objects along with the lifecycle of an operator, OLM must allow Red Hat owned operators to take advantage of Telemeter. Before moving forward, let's review the requirements that must be met for Telemeter to scrape an application's metrics:
+
+1. The application must be deployed in a namespace that is both prefixed with `openshift-` and labbeled with `openshift.io/cluster-monitoring=true`
```

Comment:
nit: labeled

Reply:
Addressed in the latest version of the PR - ran the doc through a spell checker as well.